File: backend/src/main/kotlin/com/nevexis/backend/schoolManagement/school_lessons/SchoolLesson.kt
@file:UseSerializers(
    BigDecimalSerializer::class,
    LocalDateTimeSerializer::class
)

package com.nevexis.backend.schoolManagement.school_lessons

import com.nevexis.backend.schoolManagement.school.RoomToSubjects
import com.nevexis.backend.schoolManagement.school_class.SchoolClass
import com.nevexis.backend.schoolManagement.school_period.Semester
import com.nevexis.backend.schoolManagement.school_schedule.WorkingHour
import com.nevexis.backend.schoolManagement.subject.Subject
import com.nevexis.backend.schoolManagement.users.UserView
import com.nevexis.backend.serializers.BigDecimalSerializer
import com.nevexis.backend.serializers.LocalDateTimeSerializer
import kotlinx.serialization.Serializable
import kotlinx.serialization.UseSerializers
import java.time.LocalDateTime

@Serializable
data class SchoolLesson(
    val id: Int,
    val startTimeOfLesson: LocalDateTime,
    val endTimeOfLesson: LocalDateTime,
    val subject: Subject,
    val teacher: UserView,
    val schoolClass: SchoolClass,
    val lessonTopic: String? = null,
    val room: RoomToSubjects,
    val taken: Boolean = false,
    val week: Int,
    val semester: Semester,
    val workingDay: WorkingHour,
    val status: SchoolLessonStatus
)

enum class SchoolLessonStatus {
    FREE, NORMAL, SUBSTITUTION
}

File: frontend/src/model/SchoolLesson.ts
import {Subject} from "./Subject";
import {SchoolClass} from "./SchoolClass";
import {WorkingHour} from "./PlannedSchoolLesson";
import {Semester} from "./SchoolPeriod";
import {UserView} from "./User";
import {RoomToSubjects} from "./School";

export interface SchoolLesson {
    id: number,
    startTimeOfLesson: string,
    endTimeOfLesson: string,
    subject: Subject,
    schoolClass: SchoolClass,
    lessonTopic: string | null
    room: RoomToSubjects,
    taken: boolean
    week: number,
    semester: Semester,
    workingDay: WorkingHour
    teacher: UserView,
    status: SchoolLessonStatus
}

export enum SchoolLessonStatus {
    FREE = "FREE",
    NORMAL = "NORMAL",
    SUBSTITUTION = "SUBSTITUTION"
}

export const translationOfSchoolLessonStatus = {
    FREE: "Свободен час",
    NORMAL: "Нормален",
    SUBSTITUTION: "Заместник",
}
